src/Acme/PaymentBundle/Controller/PaypalProHostedController.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Acme\PaymentBundle\Controller;
+
+use Acme\PaymentBundle\Entity\PaymentDetails;
+use Payum\Core\Payum;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Validator\Constraints\Range;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration as Extra;
+
+class PaypalProHostedController extends Controller
+{
+    /**
+     * @Extra\Route("/paypal/prohosted/prepare", name="acme_paypal_pro_hosted_prepare")
+     *
+     * @Extra\Template("AcmePaymentBundle::prepare.html.twig")
+     */
+    public function prepareAction(Request $request)
+    {
+        $gatewayName = 'paypal_pro_hosted';
+
+        $form = $this->createPurchaseForm();
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $data = $form->getData();
+
+            $storage = $this->getPayum()->getStorage(PaymentDetails::class);
+
+            /** @var $payment PaymentDetails */
+            $payment                  = $storage->create();
+            $payment['currency_code'] = $data['currency'];
+            $payment['subtotal']      = $data['amount'];
+            $payment['bn']            = 'FR_Test_H3S';
```

Comment:
It's the partner code. It's not required but recommended for a partner Paypal

bn: Identifies the source that built the code for the button. 
(Ref: Page 19 of document: https://www.paypalobjects.com/webstatic/en_GB/developer/docs/pdf/hostedsolution_uk.pdf)